### ISSUE TITLE
Missing credentials 

### ISSUE THREAD

**Issue by djkingh01** [aws-amplify/amplify-js#3030]
** Which Category is your question related to? **
Storage API, Authentication
** What AWS Services are you utilizing? **
S3
** Provide additional details e.g. code snippets **
I have a very simple use case. My users use Hosted UI for login. The user pool allows both a google as well as native cognito login with the following code. 
Auth.configure(awsmobile.Auth)

I am able to access my GraphQL services with the same. One of the Use cases requires me to access S3 and I am trying to use. 
                const template = await Storage.put(entity.displayName, entity.template, {
                    contentType:"text/html"
                })
But I get 
{"message":"Missing credentials in config","code":"CredentialsError","time":"2019-04-06T10:56:27.692Z","originalError":{"message":"Missing credentials","code":"CredentialsError","time":"2019-04-06T10:56:27.692Z"}}

What I need to do to get past this issue? I understand that I have to somehow get Identity pool credentials but when I tried to use federatedSignIn the system says you are already signedin. I need Identity pool credentails so I can directly access Amazon services from browser and I required user pool because User Pool triggers have complex rules that are built into the token for downstream access to services. 

Please help!

**Comment by powerful23** [comment]
@djkingh01 usually this is caused by duplicated packages(e.x. aws-sdk) under your `node_modules`, can you paste the `dependencies` of the `package.json` here? Also you can try `Auth.currentCredentials()` after signing in, it will return the credentials as expected.

Related #1241

**Comment by ryanjones** [comment]
I had a weird issue where I was getting this error after I had added auth through the CLI w/ federated login and then tried to change it back to just the defaults. It turns out it didn't clean up the settings in aws-exports.js for the federated login. I solved it by blowing away my auth pool and rebuilding it.

I think the error was actually stating that it didn't have the oauth provider (facebook in my case) defined in aws-exports.js (which was true).
